test/test_misc.f90

```diff
@@ -334,8 +337,87 @@ program test_misc
+  ! - of points
+  xmin = -30.0
+  xmax = 30.0
+  n_even = 4000
+  n_odd = 4001
+  ! ind_even are indices to compare in case of even number of points
+  ! ind_odd are indices to compare in case of odd number of points
+  allocate(ind_even(6),ind_odd(5))
+  ind_even = [801, 1201, 1601, 2001, 2401, 2801]
+  ind_odd = [889, 1333, 1777, 2221, 2665]
+
+  itest = itest + 1
+  test_array(itest) = testify("Hilbert transform: even function, even points")
+  allocate(x_even(n_even), hfx1_even(n_even))
+  call linspace(x_even, xmin, xmax, n_even)
+  call Hilbert_transform(fx1_array(x_even), hfx1_even)
+  call test_array(itest)%assert(hfx1_even(ind_even), hfx1_array(x_even(ind_even)), &
+       tol = 2e-4_r64)
+
+  itest = itest + 1
+  test_array(itest) = testify("Hilbert transform: odd function, even points")
+  allocate(hfx2_even(n_even))
+  call Hilbert_transform(fx2_array(x_even), hfx2_even)
+  call test_array(itest)%assert(hfx2_even(ind_even), hfx2_array(x_even(ind_even)), &
+       tol = 1e-4_r64)
+
+  itest = itest + 1
+  test_array(itest) = testify("Hilbert transform: even function, odd points")
+  allocate(x_odd(n_odd), hfx1_odd(n_odd))
+  call linspace(x_odd, xmin, xmax, n_odd)
+  call Hilbert_transform(fx1_array(x_odd), hfx1_odd)
+  call test_array(itest)%assert(hfx1_odd(ind_odd), hfx1_array(x_odd(ind_odd)), &
+       tol = 4e-4_r64)
+
+  itest = itest + 1
+  test_array(itest) = testify("Hilbert transform: odd function, odd points")
+  allocate(hfx2_odd(n_odd))
+  call Hilbert_transform(fx2_array(x_odd), hfx2_odd)
+  call test_array(itest)%assert(hfx2_odd(ind_odd), hfx2_array(x_odd(ind_odd)), &
+       tol = 1e-5_r64)
+
   tests_all = testify(test_array)
   call tests_all%report
   
   if(tests_all%get_status() .eqv. .false.) error stop -1
+
+  contains
+     ! reference functions for the Hilbert transform test
+     ! fx1 is an even function
+     function fx1_array(x) result(fx1)
+        real(r64), intent(in) :: x(:)
+        real(r64), allocatable :: fx1(:)
+        allocate(fx1(size(x)))
+        fx1 = 1/(1 + x**2)
+     end function fx1_array
+
+     ! Hfx1 is actual hilbert transform of fx1, is an odd function
+     function hfx1_array(x) result(hfx1)
+        real(r64), intent(in) :: x(:)
+        real(r64), allocatable :: hfx1(:)
+        allocate(hfx1(size(x)))
+        hfx1 = x/(1 + x**2)
+     end function hfx1_array
+
+     ! fx2 is an odd function
+     function fx2_array(x) result(fx2)
+        real(r64), intent(in) :: x(:)
+        real(r64), allocatable :: fx2(:)
+        allocate(fx2(size(x)))
+        fx2 = sin(x)/(1 + x**2)
+     end function fx2_array
+
+     ! Hfx2 is actual hilbert transform of fx2, is an even function
+     function hfx2_array(x) result(hfx2)
+        real(r64), intent(in) :: x(:)
+        real(r64), allocatable :: hfx2(:)
+        real(r64), parameter :: e = 2.718281
+        allocate(hfx2(size(x)))
+        hfx2 = (1/e - cos(x))/(1 + x**2)
```

Comment:
Check if you are having loss of precision by adding an integer to a `r64` here and elsewhere.